# Accessibility: an image button with text content does not need aria labelling. The button already has an accessible name and the aria-label or aria-labelledby will override the text content for screen reader users (`@microsoft/fluentui-jsx-a11y/icon-text-content-button-does-not-need-aria`)

<!-- end auto-generated rule header -->

Accessibility: an image button with content prop does not need aria labelling. The button already has an accessible name and the aria-label or aria-labelledby will override the text content for screen reader users.
All interactive elements must have an accessible name.

<https://www.w3.org/TR/html-aria/>

## Rule Details

This rule aims to overwrite the auto generated aria label of content prop.

Examples of **incorrect** code for this rule:

```jsx
<Button content='Close' aria-label='Submit' icon={<CloseIcon />}></Button>
```

```jsx
<Button content='Close' aria-labelledby='label-id-2' icon={<CloseIcon />}></Button>
```

Examples of **correct** code for this rule:

```jsx
<Button content="Close" icon={<CloseIcon />} />
```
